src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -95,25 +95,8 @@ int IVsEditorFactory.Close()
             // Do we need to create a text buffer?
             if (textBuffer == null)
             {
-                var contentTypeRegistryService = _componentModel.GetService<IContentTypeRegistryService>();
-                var contentType = contentTypeRegistryService.GetContentType(ContentTypeName);
-                textBuffer = editorAdaptersFactoryService.CreateVsTextBufferAdapter(_oleServiceProvider, contentType);
-
-                if (_encoding)
-                {
-                    if (textBuffer is IVsUserData userData)
-                    {
-                        // The editor shims require that the boxed value when setting the PromptOnLoad flag is a uint
-                        var hresult = userData.SetData(
-                            VSConstants.VsTextBufferUserDataGuid.VsBufferEncodingPromptOnLoad_guid,
-                            (uint)__PROMPTONLOADFLAGS.codepagePrompt);
-
-                        if (ErrorHandler.Failed(hresult))
-                        {
-                            return hresult;
-                        }
-                    }
-                }
+                textBuffer = (IVsTextBuffer)GetDocumentData(grfCreateDoc, pszMkDocument, vsHierarchy, itemid);
+                Contract.ThrowIfNull(textBuffer, $"Failed to get document data for {pszMkDocument}");
```

Comment:
This line doesn't really make sense since the GetDocumentData is non-null returning, and the cast on line 98 isn't enforcing that but again, the compiler already validated that elsewhere.